File: src/lib/types/recommendation.ts
export interface TimeRecommendation {
	id: string;
	user_id: string;
	recommendation_type: 'increase_build' | 'decrease_deep' | 'add_buffer' | 'adjust_break';
	title: string;
	description: string;
	data_source: string;
	confidence_score: number;
	is_applied: number;
	is_dismissed: number;
	created_at: string;
	applied_at: string | null;
	dismissed_at: string | null;
}

File: src-tauri/src/db/models/recommendation.rs
use serde::{Deserialize, Serialize};
use sqlx::FromRow;

#[derive(Debug, Clone, Serialize, Deserialize, FromRow)]
pub struct TimeRecommendation {
    pub id: String,
    pub user_id: String,
    pub recommendation_type: String,
    pub title: String,
    pub description: String,
    pub data_source: String,
    pub confidence_score: f64,
    pub is_applied: i32,
    pub is_dismissed: i32,
    pub created_at: String,
    pub applied_at: Option<String>,
    pub dismissed_at: Option<String>,
}

#[allow(dead_code)]
#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct CreateRecommendationInput {
    pub recommendation_type: String,
    pub title: String,
    pub description: String,
    pub data_source: String,
    pub confidence_score: f64,
}
